core/modules/system/system.module

```diff
@@ -4667,6 +4664,8 @@ function system_settings_form_submit($form, &$form_state) {
       return;
     }
 
+    // For compatibility with Drupal 7, if $form['array_filter'] is set,
+    // pass the form state values to array_filter().
```

Comment:
Proposing a bit of extra explanation to make it clear what will happen as no callback is passed to the `array_filter()` function.

```suggestion
    // For compatibility with Drupal 7, if $form['array_filter'] is set, pass
    // the form state values to array_filter(). This will remove any empty
    // values.
```